### PR TITLE
Add a RunInSpanAsync and fix namespaces

### PR DESCRIPTION
**Pull request by iantalarico**
The RunInSpanAsync fixes a bug around proper exception handling.

The namespace fix was due to a bad move in a previous PR

### CODE REVIEW

**Comment by iantalarico** [comment]
Blah broke and integration test.  Fixing now

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs [hunk 1]
```diff
@@ -139,6 +139,23 @@ public T RunInSpan<T>(Func<T> func, string name, StartSpanOptions options = null
         }
 
         /// <inheritdoc />
+        public async Task<T> RunInSpanAsync<T>(Func<Task<T>> func, string name, StartSpanOptions options = null)
+        {
+            using (StartSpan(name, options))
+            {
+                try
+                {
+                    return await func();
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs]: Please add .ConfigureAwait(false)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderPropagatingHandler.cs [hunk 2]
```diff
@@ -66,21 +66,21 @@ public class TraceHeaderPropagatingHandler : DelegatingHandler
...
             }
 
             var traceHeader = TraceHeaderContext.Create(
                 tracer.GetCurrentTraceId(), tracer.GetCurrentSpanId() ?? 0, true);
             request.Headers.Add(TraceHeaderContext.TraceHeader, traceHeader.ToString());
 
-            return await tracer.RunInSpan(
-                async () => { return await base.SendAsync(request, cancellationToken).ConfigureAwait(false); }, 
+            return tracer.RunInSpanAsync(
+                () => { return base.SendAsync(request, cancellationToken); },
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderPropagatingHandler.cs]: No need for a statement lambda - just

() => base.SendAsync(request, cancellationToken)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs]: Done

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderPropagatingHandler.cs (hunk 2 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderPropagatingHandler.cs]: Done

**Pull request merged**